src/wms/src/main/java/org/geoserver/wms/map/OpenLayersMapOutputFormat.java

```diff
@@ -287,6 +291,31 @@ private boolean hasOnlyCoverages(WMSMapContent mapContent) {
         return true;
     }
 
+    /**
+     * Helper method that checks if filtering support should be activated. If the map
+     * is not composed only by coverages then filtering should be activated. If the
+     * map is composed only of coverages but at least one of the coverages supports
+     * filtering, then filtering should be activated. Otherwise filtering capabilities
+     * will be deactivated.
+     */
+    private boolean supportsFiltering(WMSMapContent mapContent, boolean hasOnlyCoverages) {
+        return !hasOnlyCoverages || mapContent.layers().stream().anyMatch(layer -> {
```

Comment:
Filtering against non supporting layers is just ignored. What about activating if there is at least one vector layer or one supporting raster layer instead?